aspnetcore/migration/mvc.md

```diff
@@ -1,24 +1,199 @@
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk.Web">
+```
+
+For more information, see [Framework reference](xref:migration/22-to-30#framework-reference).
+
+In ASP.NET Core, the `Startup` class:
+
+* Replaces *Global.asax*.
+* Handles all app startup tasks.
+
+For more information, see <xref:fundamentals/startup>.
+
+In the ASP.NET Core project, open the *Startup.cs* file:
+
+[!code-csharp[](mvc/samples/3.x/Startup.cs?highlight=13,30,32&name=snippet)]
+
+ASP.NET Core apps must opt in to framework features with middleware. The previous template-generated code adds the following services and middleware:
+
+* The <xref:Microsoft.Extensions.DependencyInjection.MvcServiceCollectionExtensions.AddControllersWithViews*> extension method registers MVC service support for controllers, API-related features, and views. For more information on MVC service registration options, see [MVC service registration](xref:migration/22-to-30#mvc-service-registration)
+* The <xref:Microsoft.AspNetCore.Builder.StaticFileExtensions.UseStaticFiles*> extension method adds the static file handler `Microsoft.AspNetCore.StaticFiles`. The `UseStaticFiles` extension method must be called before `UseRouting`. For more information, see [Static files](xref:fundamentals/static-files).
+* The <xref:Microsoft.AspNetCore.Builder.EndpointRoutingApplicationBuilderExtensions.UseRouting*> extension method adds routing. For more information, see [Routing](xref:fundamentals/routing).
+
+This existing configuration includes what is needed to migrate the example ASP.NET MVC project. For more information on ASP.NET Core middleware options, see <xref:fundamentals/startup>.
+
+## Migrate controllers and views
+
+In the ASP.NET Core project, a new empty controller class and view class would be added to serve as placeholders using the same names as the controller and view classes in any ASP.NET MVC project to migrate from.
+
+The ASP.NET Core *WebApp1* project already includes a minimal example controller and view by the same name as the ASP.NET MVC project. So those will serve as placeholders for the ASP.NET MVC controller and views to be migrated from the ASP.NET MVC *WebApp1* project.
+
+1. Copy the methods from the ASP.NET MVC `HomeController` to replace the new ASP.NET Core `HomeController` methods. There's no need to change the return type of the action methods. The ASP.NET MVC built-in template's controller action method return type is [ActionResult](https://msdn.microsoft.com/library/system.web.mvc.actionresult(v=vs.118).aspx); in ASP.NET Core MVC, the action methods return `IActionResult` instead. `ActionResult` implements `IActionResult`.
+1. In the ASP.NET Core project, right-click the `Views` > `Home` directory, select `Add` > `Existing Item`.
```

Comment:
@scottaddie, I originally used bold ** like you are suggesting, however Rick pointed out they might be getting localized so I may have to avoid it and suggested using code ticks.  I checked, and things like **Add** **Views** were getting localized.  So I went to went to code ticks as Rick suggested.  Maybe this is something still up for more discussion?

Reply:
We do want UI element names to be localized. If, for example, I only have the Spanish language pack installed, here's what **Add** > **New Item** looks like in Spanish:

![image](https://user-images.githubusercontent.com/10702007/85324800-13480a00-b490-11ea-8486-f71eabad55ca.png)

Reply:
OK thanks!  Maybe I was misunderstanding a finer point of some sort.  @Rick-Anderson, we are all in agreement on going the bold ** route for UI elements such as **Add**, **New Item**?